### DISCUSSION TITLE
Web relay feature with multiple relay DNS names, starting in v1.0.51

### DISCUSSION THREAD

**Discussion by Ylianst** [Ylianst/MeshCentral#4249] [General]
I just published MeshCentral v1.0.51 and it has many more web relay improvements, so it should work on more web site. One thing is has this is a bit crazy is the ability to support multiple web relay DNS names at the same time.

So, if you use a web relay port or DNS name, there is a limitation that a user using one browser can only access a single remote site at any given time. If you go to a different device, hit the "HTTP" link, you just changed the web relay for the port or relay DNS and so, you lost access to the previous relayed page.

With the new version, you can do this:

```
{
  "settings": {
    "port": 443,
    "RelayDNS": [
      "relay1.myserver.com“,
      "relay2.myserver.com“,
      "relay3.myserver.com“
    ]
  }
}
```

You can set many DNS names as relays. MeshCentral will use the first one first, then the seconds, etc. If they are all in use for a user, the one that was used the longest time ago will be recycled.

Maybe this is overkill for this feature, but it does allow many relay sessions at once without having to open more TCP ports. Hopefully that makes sense.

![MC2-WebRelayMultipleDnsNames](https://user-images.githubusercontent.com/1319013/178384584-1783ae4a-45c6-4b8c-87d1-36c1ffd83203.png)

**Comment by PathfinderNetworks** [comment]
I'll try to add more DNS names in the next few days and give this a try.  I'll let you know if I run in to any hiccups.  Thanks for the work on this!

**Comment by samgaw58** [comment]
Hi I'm running mesh central 1.0.60 and seen your YouTube video on the DNS Relay. I added the line "RelayDNS" : "myrelay.example.com", restarted mesh central and it failed to start. Take the line out and everything is running again. Can you tell me what stupid thing am I doing wrong?

**Reply by si458** [reply]
make sure you have removed the `RelayPort` option, as you can only use one or the other, not both!
edit: also what is your `config.json` look like?

**Reply by samgaw58** [reply]
I hadn't tried the RelayPort option, so that isn't the problem.

**Reply by si458** [reply]
please can you share your `config.json`, blank sensitive stuff if needs be

**Reply by samgaw58** [reply]
This is my config.json that works and has been working doing all that I needed it too. Very basic I know, but like I said it does all I need it too.
Whenever I add the line **"RelayDNS" : "myrelay.example.com",** Mesh Central will fail to load. Even if I then underscore the line it still fails, the line has to be completely removed.

{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) ar>
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
"settings": {
    "cert": “mesh.example.com",
    "_WANonly": true,
    "_LANonly": true,
    "_selfupdate": "true",
    "_port": 443,
    "_clickonce": true,
    "_TlsOffload": “x”.x.x.x,
    "AgentPong": 30
  },
  "domains": {
    "": {
      "certUrl": “x”.x.x.x,
      "title2": “MySite”,
      "TitlePicture": "sitni.png",
      "loginPicture": "sitni2.png",
      "_userNameIsEmail": true
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt>
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "production": false
  }
}

**Reply by samgaw58** [reply]
This is line "certUrl": “x”.x.x.x, is "certUrl": “x.x.x.x", in the working config file.

**Reply by si458** [reply]
try adding "relayDNS" to the `settings` section?

**Reply by samgaw58** [reply]
Same outcome "relayDNS" or RelayDNS" service fails to start

**Reply by si458** [reply]
when you say 'fails to start' is the any output from running the command?

**Reply by samgaw58** [reply]
I'll put the line back in and show you the output

**Reply by samgaw58** [reply]
Would you believe me if I told you it is now working. I added the very same line that I've been trying all morning into the config.json file to get the error output for you. 
You gotta love computing...
Anyway, thank you very much for taking the time to help me out.

Best wishes.

Sam

**Reply by samgaw58** [reply]
Oh one quick question you might know the answer too. If I want to get onto HTTP on another device, where do I add the relay address. I can do it fine using router, but within the web console where is the address added?

**Reply by si458** [reply]
https://www.youtube.com/watch?v=GU7MQjkPSd8
